packages/astro/src/content/utils.ts

```diff
@@ -146,33 +183,45 @@ export function getContentEntryConfigByExtMap(settings: Pick<AstroSettings, 'con
 	return map;
 }
 
-export class NoCollectionError extends Error {}
-
-export function getEntryInfo(
-	params: Pick<ContentPaths, 'contentDir'> & {
-		entry: string | URL;
-		allowFilesOutsideCollection?: true;
+export function getEntryCollectionName({
+	contentDir,
+	entry,
+}: Pick<ContentPaths, 'contentDir'> & { entry: string | URL }) {
+	const entryPath = typeof entry === 'string' ? entry : fileURLToPath(entry);
+	const rawRelativePath = path.relative(fileURLToPath(contentDir), entryPath);
+	const collectionName = path.dirname(rawRelativePath).split(path.sep).shift() ?? '';
```

Comment:
nit: We could use [`at`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) :)

```js
path.dirname(rawRelativePath).split(path.sep).at(-1)
```

More readable IMHO

Reply:
So you like `.at` but not `??=` huh... 😉

Reply:
I am an old man, I was thought I could use - in **Java** - `[-1]` to get the last element of an array, and then JavaScript ruined everything 😆

Reply:
Ah, realized it's `[0]` actually! Probably telling that I can't remember what `shift()` vs `unshift()` is. Changed